src/openassetio-python/CMakeLists.txt

```diff
@@ -99,28 +99,68 @@ if (OPENASSETIO_ENABLE_TESTS AND Python_Interpreter_FOUND)
         COMMAND
         "${Python_EXECUTABLE}" -m venv "${CMAKE_INSTALL_PREFIX}"
         COMMAND
-        ${venv_python_dir}/python -m pip install --upgrade pip
+        ${venv_python_dir}/python -m pip install --upgrade --force-reinstall pip
     )
     # Target to wrap the venv command so it can be executed in isolation.
     add_custom_target(
         openassetio-python-venv
         DEPENDS "${installed_pip}"
     )
 
+    # One-time step to download test-specific dependencies
+    # so we don't always need an online connection for running ctest
+    set(deps_dir "${CMAKE_BINARY_DIR}/dependencies")
+    set(python_deps_dir "${deps_dir}/python")
+    set(test_deps_dir "${python_deps_dir}/tests")
+    # Store signal files under a common directory
+    set(signals_dir "${CMAKE_BINARY_DIR}/signals")
+    file(MAKE_DIRECTORY "${signals_dir}") # cmake touch doesn't create directories
+    set(test_deps_downloaded "${signals_dir}/testDepsDwnld.out")
+    add_custom_command(
+        OUTPUT "${test_deps_downloaded}"
+        COMMAND
+        "${installed_pip}" "download" "--destination-directory" "${test_deps_dir}"
+            "-r" "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+        COMMAND
+        cmake -E touch "${test_deps_downloaded}"
+        DEPENDS "${installed_pip}" "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+    )
+
+    # One-time step to download project-specific dependencies
+    set(project_deps_dir "${python_deps_dir}/project")
+    set(project_deps_downloaded "${signals_dir}/projDepsDwnld.out")
+    add_custom_command(
+        OUTPUT "${project_deps_downloaded}"
+        # Need to cache the build tools explicitely due to build isolation, see
+        # https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/#build-process
+        COMMAND
+        "${installed_pip}" wheel --wheel-dir ${project_deps_dir} setuptools wheel
+        COMMAND
+        cmake -E touch "${project_deps_downloaded}"
+        DEPENDS "${installed_pip}"
+    )
+
     # Command to install test-specific dependencies (e.g. pytest).
     add_custom_command(
         OUTPUT
         "${venv_python_dir}/pytest"
         COMMAND
-        "${installed_pip}" install -r "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
-        DEPENDS "${installed_pip}"
+        "${installed_pip}" "install" "--no-index"
+            "--find-links" "${test_deps_dir}"
+            "--force-reinstall"
+            "-r" "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+        DEPENDS "${test_deps_downloaded}"
     )
 
     # Target to install the pure Python component of the project.
     add_custom_target(
         openassetio-python-py-install
-        COMMAND "${installed_pip}" install "${PROJECT_SOURCE_DIR}"
-        DEPENDS "${installed_pip}"
+        COMMAND
+        "${installed_pip}" "install" "--no-index"
+            "--find-links" "${project_deps_dir}"
+            "--force-reinstall"
```

Comment:
[question] Do we need `--force-reinstall` here?  My understanding was that we need to force in case a `clean` had destroyed an `OUTPUT`, but there is no `OUTPUT` here.

Reply:
Addressed in https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/309/commits/77d7dd99809c24f57d8bf11163d99a1e710ff983

Removed all force installs since they don't completely fix the issues with using clean.

Reply:
:+1: to reiterate https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/309#discussion_r842652391- we added #311 to deal with this separately.